internal/functions/Connect-SqlInstance.ps1

```diff
@@ -66,7 +66,7 @@ function Connect-SqlInstance {
         [switch]$NonPooled
     )
     if ($SqlInstance.InputObject.GetType().Name -eq 'Server') {
-        if ($AzureUnsupported -and $server.DatabaseEngineType -eq "SqlAzureDatabase") {
+        if ($AzureUnsupported -and $SqlInstance.InputObject.DatabaseEngineType -eq "SqlAzureDatabase") {
```

Comment:
Not sure I see the reason to check the input object? It is known since this is an internal command where we only support the `-AzureUnsupported` when we know it can't use it.